### PR TITLE
Fix infinitely loading Of Attachment from qa.guide when opened in Search

### PR DESCRIPTION
**Pull request by shahinyan11**
### Fixed Issues

$ https://github.com/Expensify/App/issues/50006
PROPOSAL: https://github.com/Expensify/App/issues/50006#issuecomment-2392003066

### Tests

1. Go to staging.new.expensify.com
2. Go to DM
3. Send the following message. (The message is from [qa.guide@team.expensify.com](mailto:qa.guide@team.expensify.com) in #admins) 
[!(https://d2k5nsl2zxldvw.cloudfront.net/images/animations/animation_setupguide.gif)](https://camo.githubusercontent.com/6e100123165bb572e6fb35bba1dc9936d6a4691971a607ad802d6ffb61c2221a/68747470733a2f2f64326b356e736c327a786c6476772e636c6f756466726f6e742e6e65742f696d616765732f616e696d6174696f6e732f616e696d6174696f6e5f736574757067756964652e676966)
4. Go to Search > Chat
5. Click on the message sent in Step 3
5. Make sure the attachment model does not have an infinite load.

- [x] Verify that no errors appear in the JS console

### Offline tests

N/A

### QA Steps

Same as in the Tests section.
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/a0fe9b0f-b177-483f-9f64-917d00948d09

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/44390525-42c6-42c8-a345-9dbebb0430b5

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/9d988f87-230b-4dcb-b0bf-c994e53610be

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/298d28e0-925b-456b-ac53-53062c16d4ba

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/03f46d50-8acf-46dc-a73b-a61305d4dc50

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/109b72eb-9265-483e-92b5-1e81b760baec

</details>

### CODE REVIEW

File: ios/NewExpensify.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -43,7 +43,7 @@
 		D27CE6B77196EF3EF450EEAC /* PrivacyInfo.xcprivacy in Resources */ = {isa = PBXBuildFile; fileRef = 0D3F9E814828D91464DF9D35 /* PrivacyInfo.xcprivacy */; };
 		DD79042B2792E76D004484B4 /* RCTBootSplash.mm in Sources */ = {isa = PBXBuildFile; fileRef = DD79042A2792E76D004484B4 /* RCTBootSplash.mm */; };
 		DDCB2E57F334C143AC462B43 /* ExpoModulesProvider.swift in Sources */ = {isa = PBXBuildFile; fileRef = 4D20D83B0E39BA6D21761E72 /* ExpoModulesProvider.swift */; };
-		E51DC681C7DEE40AEBDDFBFE /* (null) in Frameworks */ = {isa = PBXBuildFile; };
+		E51DC681C7DEE40AEBDDFBFE /* BuildFile in Frameworks */ = {isa = PBXBuildFile; };
```
Comment (suneox) [inline ios/NewExpensify.xcodeproj/project.pbxproj:46]: Please revert the changes for this file.​

**Comment by suneox** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Comment by suneox** [comment]
@shahinyan11 Please fix the ESLint check

**Comment by shahinyan11** [comment]
> @shahinyan11 Please fix the ESLint check

ESLint error related to `useOnyx` instead of `withOnyx`. Shouldn't this be done as a separate task?

**Comment by suneox** [comment]
> @suneox ESLint error related to using `useOnyx` instead of `withOnyx`. Shouldn't this be done as a separate task?

Ah, this error is not related to the scope change of this issue, we can skip this one.

**Comment by shahinyan11** [comment]
@suneox Is there anything else I need to do?

**Comment by suneox** [comment]
> @suneox Is there anything else I need to do?

@shahinyan11 Nope, another refactor will handle ESLint for this one.

**Review by suneox — APPROVED** [review]
The pipeline fails due to the changed files, and the scope of this issue is not related to `withOnyx`, so we can handle replace `withOnyx` in another task

File: src/ROUTES.ts [hunk 2]
```diff
@@ -303,8 +303,12 @@ const ROUTES = {
     },
     ATTACHMENTS: {
         route: 'attachment',
-        getRoute: (reportID: string, type: ValueOf<typeof CONST.ATTACHMENT_TYPE>, url: string, accountID?: number) =>
-            `attachment?source=${encodeURIComponent(url)}&type=${type}${reportID ? `&reportID=${reportID}` : ''}${accountID ? `&accountID=${accountID}` : ''}` as const,
+        getRoute: (reportID: string, type: ValueOf<typeof CONST.ATTACHMENT_TYPE>, url: string, accountID?: number, isAuthTokenRequired?: boolean) => {
+            const reportParam = reportID ? `&reportID=${reportID}` : '';
+            const accountParam = accountID ? `&accountID=${accountID}` : '';
+            const authParam = isAuthTokenRequired ? '&isAuthTokenRequired=true' : '';
+            return `attachment?source=${encodeURIComponent(url)}&type=${type}${reportParam}${accountParam}${authParam}` as const;
```
Comment (cead22) [inline src/ROUTES.ts:310]:
```suggestion
            const authTokenParam = isAuthTokenRequired ? '&isAuthTokenRequired=true' : '';
            return `attachment?source=${encodeURIComponent(url)}&type=${type}${reportParam}${accountParam}${authTokenParam}` as const;
```

**Review by cead22 — CHANGES_REQUESTED** [review]

**Comment by cead22** [comment]
Reminder from the updated contributing guidelines

>[Open a pull request](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork). It is required to complete every step and check every box in the PR Author Checklist. If a box has been checked without the action being taken, it will be a violation of [Rule #2](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) and could lead to [a warning](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md#compliance) being issued.

This item is checked and there's a warning on videos in both the PR author and reviewer checklists. Can you please report it or open an issue for it and link it here?

> * I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)

**Comment by suneox** [comment]
@shahinyan11 could you please verify the author checklist item `I verified there are no console errors`? Last time, I only saw a pipeline error due to the `Changed files ESLint check` 🤔. Did we have any issue for this item?

**Comment by cead22** [comment]
Let me know if there are more recent guidelines for updating withOnyx with useOnyx, but from [this post](https://expensify.slack.com/archives/C01GTK53T8Q/p1726169039084589), I think we should include those updates in this PR

**Comment by cead22** [comment]
> @shahinyan11 could you please verify the author checklist item `I verified there are no console errors`? Last time, I only saw a pipeline error due to the `Changed files ESLint check` 🤔. Did we have any issue for this item?

I unchecked this in both checklists because I saw console errors, and didn't see where those were fixed or reported

**Comment by suneox** [comment]
> Let me know if there are more recent guidelines for updating withOnyx with useOnyx, but from [this post](https://expensify.slack.com/archives/C01GTK53T8Q/p1726169039084589), I think we should include those updates in this PR

@cead22 I got it, thank you for pointing this out! 

@shahinyan11 We will handle pipeline in this PR

**Comment by shahinyan11** [comment]
@suneox @cead22 What console error are you talking about? Do you mean `TRenderEngineProvider: Support for defaultProps` ?

**Comment by cead22** [comment]
> @suneox What console error are you talking about? Do you mean `TRenderEngineProvider: Support for defaultProps` ?

Yes, that one on the Android Native video on the submitter checklist and the iOS native on the reviewer checklist. And `Function components cannot be given ref...` on the iOS native video of the reviewer checklist

> And about migration to `useOnex` I have a doubts. We have a logic in `memo` which depends on `props.user`. is it worth deleting `withOnyx` in this case ?

I'm not sure actually, can you ask in #expensify-open-source please?

**Comment by shahinyan11** [comment]
@suneox @cead22 I have added discussion in [Slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1728466968987479) about ESLint error, but I do not know whose answer should I follow

**Comment by shahinyan11** [comment]
> Yes, that one on the Android Native video on the submitter checklist and the iOS native on the reviewer checklist. And `Function components cannot be given ref...` on the iOS native video of the reviewer checklist
ure actually, can you ask in #expensify-open-source please?

@cead22 That warning exists in the `main` branch and does not appear from my changes. What should I do in this case

**Comment by suneox** [comment]
> @cead22 That warning exists in the `main` branch and does not appear from my changes. What should I do in this case

Regarding the console errors that were raised previously, and we have other issues being addressed in #49062 and #49035

**Comment by suneox** [comment]
> @suneox @cead22 I have added discussion in [Slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1728466968987479) about ESLint error, but I do not know whose answer should I follow

@shahinyan11 I’ve provided a solution in your [thread](https://expensify.slack.com/archives/C01GTK53T8Q/p1728466968987479), and I’m also waiting for confirmation in the C+ [thread](https://expensify.slack.com/archives/C02NK2DQWUX/p1728479035439009)

**Comment by shahinyan11** [comment]
> @shahinyan11 I’ve provided a solution in your [thread](https://expensify.slack.com/archives/C01GTK53T8Q/p1728466968987479), and I’m also waiting for confirmation in the C+ [thread](https://expensify.slack.com/archives/C02NK2DQWUX/p1728479035439009)

Should I implement your solution now or wait for confirmation?

**Comment by suneox** [comment]
> Should I implement your solution now or wait for confirmation?

I’m still waiting for feedback, let’s wait for a confirmation.

**Comment by shahinyan11** [comment]
@cead22 Could I set this point as checked without any additional comments  
> I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)

since @suneox  reported it in this [comment](https://github.com/Expensify/App/pull/50427#issuecomment-2402359170)

**Comment by cead22** [comment]
Yup, that's great. The process works :)

**Comment by cead22** [comment]
@suneox can you give this another review please, since a few things changed? Thanks

**Comment by suneox** [comment]
@shahinyan11 Could you please sync with the latest main? I’d like to verify the re-render after migrate to useOnyx. Thanks!

File: src/components/HTMLEngineProvider/HTMLRenderers/ImageRendererWrapper.tsx
Comment (suneox) [inline src/components/HTMLEngineProvider/HTMLRenderers/ImageRendererWrapper.tsx:1]: We don’t need to create a new file, we should just keep it in the `ImageRenderer` component as this pattern doesn’t exist in the code base

File: src/components/HTMLEngineProvider/HTMLRenderers/ImageRenderer.tsx [hunk 3]
```diff
@@ -113,14 +111,4 @@ function ImageRenderer({tnode}: ImageRendererProps) {
...
-    user: {
-        key: ONYXKEYS.USER,
-    },
-})(
-    memo(
-        ImageRenderer,
-        (prevProps, nextProps) => prevProps.tnode.attributes === nextProps.tnode.attributes && prevProps.user?.shouldUseStagingServer === nextProps.user?.shouldUseStagingServer,
-    ),
-);
+export default ImageRenderer;
```
Comment (suneox) [inline src/components/HTMLEngineProvider/HTMLRenderers/ImageRenderer.tsx:114]: Please move the logic from ImageRendererWrapper file to this file
```suggestion
export default ImageRendererWrapper;
```

File: src/components/HTMLEngineProvider/HTMLRenderers/ImageRenderer.tsx [hunk 4]
```diff
@@ -1,6 +1,5 @@
 import React, {memo} from 'react';
-import {withOnyx} from 'react-native-onyx';
-import type {OnyxEntry} from 'react-native-onyx';
+import {OnyxEntry, useOnyx} from 'react-native-onyx';
```
Comment (suneox) [inline src/components/HTMLEngineProvider/HTMLRenderers/ImageRenderer.tsx]: Please update to import type
```suggestion
import type {OnyxEntry} from 'react-native-onyx';
import {useOnyx} from 'react-native-onyx';
```

**Review by suneox — APPROVED** [review]
I have verified that migrating to useOnyx keeps the render times the same for ImageRenderer component

**Comment by suneox** [comment]
> I have verified that migrating to useOnyx keeps the render times the same for ImageRenderer component

@cead22 the new change has been reviewed

**Review by cead22 — APPROVED** [review]

**Pull request merged**